IF/src/main/java/com/github/stefvanschie/inventoryframework/gui/GuiListener.java

```diff
@@ -191,19 +194,16 @@ public void resetItemsSmithingTable(@NotNull InventoryClickEvent event) {
      * @param event the event fired when an entity picks up an item
      * @since 0.6.1
      */
+    @SuppressWarnings("deprecation")
     @EventHandler(ignoreCancelled = true, priority = EventPriority.MONITOR)
-    public void onEntityPickupItem(@NotNull EntityPickupItemEvent event) {
-        if (!(event.getEntity() instanceof HumanEntity)) {
-            return;
-        }
-
-        Gui gui = getGui(((HumanEntity) event.getEntity()).getOpenInventory().getTopInventory());
+    public void onEntityPickupItem(@NotNull PlayerPickupItemEvent event) {

```

Comment:
Wouldn’t it be better to create such event methods in a separate class according to version support? Thus, there would be no future problems in version 1.17, for example.